src/clients.rs

```diff
@@ -34,14 +34,17 @@ pub async fn create_trunk_repo(
         .await
     {
         Ok(resp) => resp,
-        Err(e) => return Err(anyhow::anyhow!(e).context("Failed to create trunk repo")),
+        Err(e) => return Err(anyhow::anyhow!(e).context("Failed to validate trunk repo")),
     };
 
-    if resp.status() != reqwest::StatusCode::OK {
-        return Err(
-            anyhow::anyhow!("{}: {}", resp.status(), status_code_help(resp.status()))
-                .context("Failed to create trunk repo"),
-        );
+    if resp.status() != reqwest::StatusCode::OK
+        && resp.text().await?.contains("Organization not found")
```

Comment:
Should we check with status `404` instead of text?

Reply:
Yeah, it seems like the only case where we return 404 from the public_api endpoints is if org isn't found from orgslug

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

(1) If status is 5xx the issue is on our side,  we should explain what happened, recommend actions and return 0.
(2) If status is 4xx the issue is on client side, we should explain what happened, recommend actions and return 1.
(3) Otherwise, if the response is unexpected handle it as (1).